### ISSUE TITLE
support SPEAR-TTS

### ISSUE THREAD

**Issue by Rongjiehuang** [lucidrains/audiolm-pytorch#84]
Hi author, thanks for your contribution!

Google has released their new work named SPEAR-TTS (https://arxiv.org/abs/2302.03540), whose arch is similar to `audiolm`. Please consider interpolating it into this repo, and that will be really helpful to the community. Thanks!

**Comment by lucidrains** [comment]
@Rongjiehuang yes, that seems like a good paper, i will definitely consider it

**Comment by lucidrains** [comment]
maybe audio is going to be a solved problem in a year or two

things are happening fast

**Comment by Approximetal** [comment]
> took a look at this paper, and i think it may deserve its own repository. S1 in the paper with the pretraining and backtranslation may be too much to add

I agree.

**Comment by lucidrains** [comment]
@Approximetal i think what i'll do is build out the text-to-semantic module over at https://github.com/lucidrains/soundstorm-pytorch

**Comment by abhiprojectz** [comment]
> @Approximetal i think what i'll do is build out the text-to-semantic module over at https://github.com/lucidrains/soundstorm-pytorch

Yes, That will also help in cloning voices using Bark whose arch is also same as AudioLM , only text missing is semantic tokens from audio extraction.

**Comment by lucidrains** [comment]
@abhiprojectz ok nice, will be traveling soon, but should resume work mid next month and get this done by end of June

**Comment by lucidrains** [comment]
i haven't even read the paper yet, and i'm guessing it is nothing more than an attention net, maybe with some CTC loss involved?

**Comment by lucidrains** [comment]
embarking on the project here https://github.com/lucidrains/spear-tts-pytorch
